shared/actions/searchv3/index.js

```diff
@@ -0,0 +1,64 @@
+// @flow
+import * as Constants from '../../constants/searchv3'
+import * as Creators from './creators'
+import * as EntityAction from '../entities'
+import {List} from 'immutable'
+import {apiserverGetWithSessionRpc} from '../../constants/types/flow-types'
+import {trim, keyBy} from 'lodash'
+import {call, put, select} from 'redux-saga/effects'
+import * as Selectors from '../../constants/selectors'
+import * as Saga from '../../util/saga'
+
+import type {SagaGenerator} from '../../constants/types/saga'
+
+function _apiSearch(searchTerm: string, service: string = '', limit: number = 20) {
+  return new Promise((resolve, reject) => {
```

Comment:
we already have promise helpers for this generated right? https://github.com/keybase/client/blob/master/protocol/js/flow-types.js#L1082